api_tests/nodes/views/test_node_list.py

```diff
@@ -385,6 +406,39 @@ def test_incorrect_filtering_field_not_logged_in(self):
         assert_equal(len(errors), 1)
         assert_equal(errors[0]['detail'], "'notafield' is not a valid field for this endpoint.")
 
+    def test_filtering_on_root(self):
+        root = ProjectFactory(is_public=True)
+        child = ProjectFactory(parent=root, is_public=True)
+        ProjectFactory(parent=root, is_public=True)
+        ProjectFactory(parent=child, is_public=True)
+        # create some unrelated projects
+        ProjectFactory(title="Road Dogg Jesse James", is_public=True)
+        ProjectFactory(title="Badd *** Billy Gunn", is_public=True)
+
+        url = '/{}nodes/?filter[root]={}'.format(API_BASE, root._id)
+
+        res = self.app.get(url, auth=self.user_one.auth)
+        assert_equal(res.status_code, 200)
+        assert_equal(res.json['links']['meta']['total'], 4)
+
+    def test_filtering_on_null_parent(self):
```

Comment:
I wouldn't be surprised if there isn't. We aren't testing everything filtering related, just some general example and new or exceptional cases. Testing filtering on root covers the "relationship link" filtering, and null parent tests the filtering on a null relationship link.